### PR TITLE
Port git-dit-tag

### LINKED ISSUE #50: Port "git-dit-tag" to rust
What the title says.

### PR DESCRIPTION
**Pull request by neithernut**
Resolves #50.

### CODE REVIEW

**Comment by neithernut** [comment]
Still needs some testing.

**Comment by neithernut** [comment]
Superficially tested.

File: lib/src/repository.rs [hunk 1]
```diff
@@ -80,6 +84,13 @@ impl RepositoryExt for Repository {
             .chain_err(|| EK::WrappedGitError)
     }
 
+    fn get_local_issue_head(&self, issue: Oid) -> Result<Reference> {
+        let glob = format!("refs/dit/{}/head", issue);
+        self.references_glob(&glob).chain_err(|| EK::WrappedGitError)?
+            .next().ok_or_else(|| Error::from_kind(EK::CannotFindIssueHead(issue)))
+            .and_then(|reference| reference.chain_err(|| EK::WrappedGitError))
```
Comment (matthiasbeyer) [inline lib/src/repository.rs]: I would really like to see each method call on one line. Its way more readable.

File: src/main.rs [hunk 2]
```diff
@@ -306,6 +298,57 @@ fn reply_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
     0
 }
 
+/// reply subcommand implementation
```
Comment (matthiasbeyer) [inline src/main.rs]: Nope!

File: src/main.rs [hunk 3]
```diff
@@ -306,6 +298,57 @@ fn reply_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
...
+///
+fn tag_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    // NOTE: the issue-hash is a required parameter
+    let issue = try_or_1!(Oid::from_str(matches.value_of("issue-hash").unwrap()));
+
+    // get the head for the issue to tag
+    let mut issue_head = {
+        try_or_1!(repo.get_local_issue_head(issue))
+    };
```
Comment (matthiasbeyer) [inline src/main.rs]: No need for this scope, right?

File: src/util.rs [hunk 4]
```diff
@@ -93,6 +97,12 @@ impl<'r> RepositoryUtil<'r> for Repository {
                .unwrap_or_else(|| self.path().join("COMMIT_EDITMSG"))
     }
 
+    fn cli_references(&'r self, matches: &ArgMatches) -> Result<Vec<Commit<'r>>> {
+        matches.values_of("reference")
+               .map(|p| self.values_to_hashes(p))
+               .unwrap_or(Ok(Vec::new()))
```
Comment (matthiasbeyer) [inline src/util.rs]: While you're at it: `Ok(vec![])` looks more natural to Rustaceans IMO.

File: src/main.rs (hunk 2 above)
Comment (neithernut) [inline src/main.rs]: Good catch

**Pull request merged**